utilities/blob_db/blob_db_impl.cc

```diff
@@ -926,15 +926,41 @@ uint64_t BlobDBImpl::ExtractExpiration(const Slice& key, const Slice& value,
   return has_expiration ? expiration : kNoExpiration;
 }
 
+std::shared_ptr<BlobFile> BlobDBImpl::GetOldestBlobFile() {
+  std::vector<std::shared_ptr<BlobFile>> blob_files;
+  CopyBlobFiles(&blob_files);
+  if (blob_files.empty()) {
+    return nullptr;
+  }
+  blobf_compare_ttl compare;
```

Comment:
You also need to filter out files that are either `!Immutable()` (still open for writes) or `Obsolete()` (pending delete already).

Reply:
Ah, good point. I'll make the change.